### ISSUE TITLE
FortinetFortiOSDriver not working for me

### ISSUE THREAD

**Issue by salvapinyol** [scrapli/scrapli_community#161]
**Describe the bug**
I cannot connect to a fortinet firewall using FortinetFortiOSDriver. I got error: 

`CRITICAL:scrapli.transport:operation timed out, closing connection_`

**To Reproduce**
Authentication works ok, but it seems that the comms_prompt_pattern is not working, I've tried to modify it but without success.

**Expected behavior**
connection shoud work.

**Stack Trace**
Here you can find the scrapli debug log:

```
DEBUG:scrapli:Scrapli factory initialized
INFO:scrapli:Driver '<class 'scrapli_community.fortinet.fortios.sync_driver.FortinetFortiOSDriver'>' selected from scrapli community platforms, with the following platform arguments: '{'comms_prompt_pattern': '[\\w_-]+ (\\(\\w+\\) )?[$#]', 'on_open': <function default_sync_on_open at 0x7f623a267430>, 'on_close': <function default_sync_on_close at 0x7f623a267a60>}'
DEBUG:scrapli.driver:load core transport requested
DEBUG:scrapli.driver:core transport 'system' loaded successfully
INFO:scrapli.driver:opening connection to 'SATOFCN2' on port '22'
DEBUG:scrapli.transport:opening transport connection to 'SATOFCN2' on port '22'
DEBUG:scrapli.transport:created transport 'open_cmd': '['ssh', 'SATOFCN2', '-p', '22', '-o', 'ConnectTimeout=15', '-o', 'ServerAliveInterval=30', '-l', 'sys-ft-sa0001', '-o', 'StrictHostKeyChecking=no', '-o', 'UserKnownHostsFile=/dev/null', '-F', '/dev/null', '-o', 'PubkeyAuthentication=no']'
DEBUG:scrapli.transport:transport connection to 'SATOFCN2' on port '22' opened successfully
INFO:scrapli.channel:channel log enabled, logging channel output to '/home/spinyol/logs/nwcli/SATOFCN2-scrapli-2023-11-24:17.22.21.log'
DEBUG:scrapli.channel:attempting in channel ssh authentication
DEBUG:scrapli.channel:read: b"Warning: Permanently added 'satofcn2,10.148.120.130' (ED25519) to the list of known hosts.\n"
DEBUG:scrapli.channel:read: b"sys-ft-sa0001@satofcn2's password: "
DEBUG:scrapli.channel:write: REDACTED
DEBUG:scrapli.channel:write: '\n'
DEBUG:scrapli.channel:read: b'\n'
DEBUG:scrapli.channel:read: b'SATOFCN2 $ '
CRITICAL:scrapli.transport:operation timed out, closing connection
DEBUG:scrapli.transport:closing transport connection to 'SATOFCN2' on port '22'
DEBUG:scrapli.transport:transport connection to 'SATOFCN2' on port '22' closed successfully
```

**Screenshots**
N/A

**OS (please complete the following information):**
 - OS: redhat enterprise linux
 - scrapli version - 2023.7.30
 - scrapli_community version - 2023.7.30
 - version of any optional extras (paramiko|ssh2-python|textfsm, etc.)  - N/A

**Additional context**
FORTIOS versions tested: 6.4.14 and 7.2.5

**Comment by carlmontanari** [comment]
> Authentication works ok, but it seems that the comms_prompt_pattern is not working, I've tried to modify it but without success.

hmm, just looking quickly looks like `SATOFCN2 $ ` would be matched. I don't have anything forti to investigate so you are probably mostly on your own. you may want to look in [here](https://github.com/scrapli/scrapli_community/blob/main/scrapli_community/fortinet/fortios/sync_driver.py#L37) to see if anything else in there is causing an issue and/or at least use that as a jumping off point to troubleshoot.

**Comment by salvapinyol** [comment]
hi @carlmontanari . 

Thanks a lot for your suggestion. 
If I remove these lines from prepare_session() method:  [here](https://github.com/scrapli/scrapli_community/blob/main/scrapli_community/fortinet/fortios/sync_driver.py#L37)
```
        initial = self.channel.read()          
        if "(Press 'a' to accept):" in str(initial):
            self.channel.write("a")**
```
it works fine for me. But doing this will broke another deployments where _set post-login-banner enable_ is configured on the firewall.

Unfortunately, I don't know how to fix this to make it work for both cases. There is a way to check if there is something pending to read in the channel before doing the channel.read() ?

**Comment by carlmontanari** [comment]
> a way to check if there is something pending to read in the channel before doing the channel.read()

nope, that's part of the fun of ssh stuff 🤣 -- but you can probably have a `open_session_buf` var, read off the channel, if that stuff you removed is in the read output, then do whatever you need to do, and otherwise continue on *but* subsequent reads first read out of the `open_session_buf`. you can look at scrapli-netconf we do basically the same thing there.

like I said though you're going to be on your own for this since I don't have any forti stuff. moreover you're going to have to probably have some good tests in place for me to merge anything since ill have to have faith that you got it sorted out well 😁 !

alternatives could be: create a `variant` (not sure if that will be enough to handle it), create an entirely different platform, write a custom `on_open` that handles this in your own way (this is probably the most "right" way, and you can probably do what I described above w/ a temporary buf to put things you read off the channel until you figure out what you need to do to get fully established)

**Comment by carlmontanari** [comment]
gunna close this for now. I still think the custom on open func is probably the "most right" way to sort this out -- hope that worked for whoever sees this :D
